### PR TITLE
(demo): update prometheus to use otlp write

### PR DESCRIPTION
**Pull request by puckpuck**
Updates Prometheus to use OTLP write. Collector exporter updated accordingly.

References this [demo repo PR](https://github.com/open-telemetry/opentelemetry-demo/pull/1149)

### CODE REVIEW

**Review by TylerHelmuth — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
Do we need to wait for the demo PR to merge before merging this?

**Review by reyang — APPROVED** [review]

**Comment by puckpuck** [comment]
Demo repo is merged. We can merge this too now.

**Pull request merged**
